celery/bin/worker.py

```diff
@@ -42,6 +75,8 @@ def __init__(self):
     def convert(self, value, param, ctx):
         # Pools like eventlet/gevent needs to patch libs as early
         # as possible.
+        maybe_patch_concurrency(value)
```

Comment:
I'm pretty sure that's the wrong hook for this.

Reply:
I'm open for suggestions.
My reasoning was to make it as close as possible to `-P` evaluation which is the only indication that patching is needed.

A possibility is to move this call to the command function itself, but then we can't reference the app in any of the `click.option`s (bc it will cause the app to be imported and the patching to be too late)